android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java

```diff
@@ -65,28 +67,48 @@ private static String getApiKey(Context context){
     return resourceId != 0 ? context.getString(resourceId) : "";
   }
 
+  private static String getCurrentPwsEndpoint(Context context){
+    String defaultEndpoint = isGoogleApiKeyAvailable(context) ?
+      convertToSettingsEndpointFormat(GOOGLE_ENDPOINT, GOOGLE_ENDPOINT_VERSION) :
+      convertToSettingsEndpointFormat(PROD_ENDPOINT, PROD_ENDPOINT_VERSION);
+    SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(context);
+    return sharedPref.getString(context.getString(R.string.pws_endpoint_setting_key),
+                                defaultEndpoint);
+  }
+
+  private static String getCurrentPwsEndpointUrl(Context context){
+    String endpoint = getCurrentPwsEndpoint(context);
+    return endpoint.split(",")[0];
+  }
+
+  private static int getCurrentPwsEndpointVersion(Context context){
+    String endpoint = getCurrentPwsEndpoint(context);
+    return Integer.parseInt(endpoint.split(",")[1]);
+  }
+
+  public static String convertToSettingsEndpointFormat(String pwsUrl, int pwsVerion){
+    return pwsUrl + "," + pwsVerion;
+  }
+
   public static void setPwsEndpoint(Context context, PwsClient pwsClient){
     String apiKey = getApiKey(context);
-    String pwsEndpoint = PROD_ENDPOINT;
-    int apiVersion = PROD_ENDPOINT_VERSION;
-    if (!apiKey.isEmpty()){
-      pwsEndpoint = GOOGLE_ENDPOINT;
-      apiVersion = GOOGLE_ENDPOINT_VERSION;
-    }
+    String pwsEndpoint = getCurrentPwsEndpointUrl(context);
+    int apiVersion = getCurrentPwsEndpointVersion(context);
     pwsClient.setEndpoint(pwsEndpoint, apiVersion, apiKey);
   }
 
   public static void setPwsEndpoint(Context context, PhysicalWebCollection physicalWebCollection){
     String apiKey = getApiKey(context);
-    String pwsEndpoint = PROD_ENDPOINT;
-    int apiVersion = PROD_ENDPOINT_VERSION;
-    if (!apiKey.isEmpty()){
-      pwsEndpoint = GOOGLE_ENDPOINT;
-      apiVersion = GOOGLE_ENDPOINT_VERSION;
-    }
+    String pwsEndpoint = getCurrentPwsEndpointUrl(context);
+    int apiVersion = getCurrentPwsEndpointVersion(context);
     physicalWebCollection.setPwsEndpoint(pwsEndpoint, apiVersion, apiKey);
   }
 
+  public static boolean isGoogleApiKeyAvailable(Context context){
+    return context.getResources().getIdentifier("google_api_key", "string",
```

Comment:
Can you make a private `getGoogleApiKeyResourceId` that this and `getApiKey` uses?